doc/developers.md

```diff
@@ -0,0 +1,73 @@
+# Developer documentation
```

Comment:
If this site has API docs, then won't users use it? Maybe in a future PR "Developer documentation" can become a sub-page of the whole site.

Reply:
Depends. Users who just want to compile this then run their MATLAB scripts, no. Although how many of them are we expecting? Unsure.

Reply:
👍 fair enough - I don't know how users interact with the code so wasn't sure if the API docs was more dev or user focused, but it sounds like dev and therefore this is fine.

Reply:
> Maybe in a future PR "Developer documentation" can become a sub-page of the whole site.

I had been thinking of this though. So I'm glad I'm not the only one.

Reply:
as far as I'm aware the only user interaction with the c++ code is `tdms <input_file> <output_file>`